src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -145,10 +149,20 @@ public override void Log(ILogMessage logMessage)
                 // Avoid moving to the UI thread unless there is work to do
                 if (reportProgress || showAsOutputMessage)
                 {
+                    // Make sure the message is queued in order of calls to LogAsync, but don't wait for the UI thread
+                    // to actually show it.
+                    _loggedMessages.Enqueue(Tuple.Create(reportProgress, showAsOutputMessage, logMessage));
```

Comment:
@benaadams I predict that in the near future someone will finally add `System.ValueTuple` to this project and go through the code searching for `Tuple.Create`. The currently implementation works well with such a transition.